### PR TITLE
Dk Diseases +15% (Patch 3.2) / Sanctified Wrath / Scourge Strike / Wandering Plague / Renewed Hope + Flash heal

### PR DESCRIPTION
**Pull request by P-Kito**

### CODE REVIEW

**Comment by P-Kito** [comment]
Wtf how did this thing go in there <_<

**Comment by P-Kito** [comment]
Unfortunately it made one PR for 2 things.

**Comment by P-Kito** [comment]
Unfortunately it made one PR for 3 things.

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -10278,6 +10278,11 @@ float Unit::SpellDamagePctDone(Unit* victim, SpellInfo const* spellProto, Damage
                 if (AuraEffect* aurEff = GetDummyAuraEffect(SPELLFAMILY_DEATHKNIGHT, 196, 0))
                     if (victim->GetDiseasesByCaster(owner->GetGUID()) > 0)
                         AddPct(DoneTotalMod, aurEff->GetAmount());
+
+            // Diseases should make 15% more damage with patch 3.2.0
+            if (spellProto->SpellFamilyFlags[1] & 0x6000000 && spellProto->SpellFamilyFlags[1] & 0x800)
```
Comment (Kittnz) [inline src/server/game/Entities/Unit/Unit.cpp]: no name for 0x6000000 ?

**Comment by Nyeriah** [comment]
This PR can't be reviewed as it is. Please create a new branch for each PR so it does not include unwanted changes.

File: src/server/scripts/Spells/spell_dk.cpp [hunk 2]
```diff
@@ -1445,6 +1445,7 @@ class spell_dk_scourge_strike : public SpellScriptLoader
 
         private:
             float multiplier;
+            int32 damageDone;
```
Comment (Nyeriah) [inline src/server/scripts/Spells/spell_dk.cpp]: int32 damageDone is left uninitialized.

File: src/server/game/Entities/Unit/Unit.cpp [hunk 3]
```diff
@@ -1647,6 +1647,9 @@ void Unit::CalcAbsorbResist(Unit* victim, SpellSchoolMask schoolMask, DamageEffe
 
     RoundToInterval(auraAbsorbMod, 0.0f, 100.0f);
 
+    auto absorb_ignoring_damage = CalculatePct(dmgInfo.GetDamage(), auraAbsorbMod);
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.cpp]: please don't use the auto keyword for such simple values as the return type of CalculatePct()

**Comment by jackpoz** [comment]
This PR has the purpose of fixing some spells but it modifies code that affects other spells too. Please verify that all other spells are working correctly or at least that no new issues are added.

**Comment by P-Kito** [comment]
Which other spells does it effect? The Absorb-stuff is only for sanctified wrath, since that's pretty much the only spell with this spell_aura_mod_.... 

I see no reason to not use the auto keyboard there.

It was enough work to create this PR. It's working for me as it should. Maybe this brings some ideas for the recent contributors to make a PR of your dreams.

**Comment by tkrokli** [comment]
I agree. This PR has got 5 different fixes mixed into 1 PR. It should have been 5 PRs, 1 for each commit.

**Comment by P-Kito** [comment]
I dont even know how to do this, since every commit I did was just attached to this :/
Well, sorry I was b1tching around on my comment. Probaby someday I will change it... or someone else :)

**Comment by tkrokli** [comment]
@P-Kito : it should be simple enough.
If you just use git or GitExtensions to create a new **branch** based on an updated straight TC 3.3.5 source.

Name them something short, like:
- "CalcAbsorbResist"
- "dk_disease_dmg"
- "dk_ss_shadow_dmg"
- "dk_spell_wp_pvp"
- "priest_custom_crit_rh_fh"

Switch to each branch. Apply only 1 of the 5 first commits in this PR to each of those branches. :)

**Comment by ccrs** [comment]
@tkrokli it's simple, anyone can do it. Just takes too much time :/

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (ccrs) [inline src/server/game/Entities/Unit/Unit.cpp]: there is no definition for spellfamilyflags

**Comment by tkrokli** [comment]
I got the time, if P-Kito does not feel like doing it. It's just that I'm not good at coding.

**Comment by P-Kito** [comment]
Go ahead if you like to do it. I appreciate that you take the time.

**Comment by tkrokli** [comment]
OK, very well. If needed, I can add collaborator access to my branches. I'll do it as I wrote above.

**Comment by P-Kito** [comment]
Thank you VERY VERY MUCH. 
Let me know if I can do anything for you.

**Comment by tkrokli** [comment]
Well, there is the issue of hardcoded values in https://github.com/TrinityCore/TrinityCore/pull/15037/files
I was looking for a way to replace
```cpp
if (AuraEffect* aurEff = GetAuraEffectOfRankedSpell(57470, EFFECT_0))
```
with
```cpp
if (AuraEffect* aurEff = GetAuraEffectOfRankedSpell(SPELL_PRIEST_RENEWED_HOPE, EFFECT_0))
```
but that would return spell ID `63944` instead of `57470`. Any idea where to look up a name for `spell 57470` or if it needs to be created in spell_priest.cpp or something like that? (+ similar for `SpellIconID 177`).

**Comment by ccrs** [comment]
there are also no definitions for spelliconid

**Comment by Aokromes** [comment]
#15033 #15034 #15035 #15036 #15037

**Pull request closed without merging**